src/components/FileUploader/useIsCurrentlyDragging.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useEffect, useState } from "react"
+
+let draggingCount = 0
```

Comment:
oh yeah! `useRef` is a great idea. It's basically a count of how many times the drag events are recorded. Keeping track with `0` or `1` is a way to prevent the callbacks from running in a loop and causing bugs.